rsconnect/api.py

```diff
@@ -292,21 +292,26 @@ def get_content(self, content_guid):
         return results
 
     def wait_for_task(
-        self, task_id, log_callback, abort_func=lambda: False, timeout=get_timeout(), poll_wait=0.5, raise_on_error=True
+        self,
+        task_id,
+        log_callback,
+        abort_func=lambda: False,
+        timeout=get_task_timeout(),
+        poll_wait=0.5,
+        raise_on_error=True,
     ):
-        last_status = None
-        ending = time.time() + timeout if timeout else 999999999999
-
         if log_callback is None:
             log_lines = []
             log_callback = log_lines.append
         else:
             log_lines = None
 
+        last_status = None
+        start_time = time.time()
         sleep_duration = 0.5
         time_slept = 0
         while True:
-            if time.time() >= ending:
+            if time.time() - start_time < timeout:
```

Comment:
This is backwards, yes? As written, you're saying "if elapsed time is less than the timeout, err". Since timeout defaults to maxsize, that will always be true.

Do you have a test to confirm this reading?